main.go

```diff
@@ -255,25 +280,27 @@ func run() int {
 				http.Error(w, fmt.Sprintf("failed to reload config: %s", err), http.StatusInternalServerError)
 			}
 		})
-	http.Handle("/metrics", promhttp.Handler())
-	http.HandleFunc("/probe", func(w http.ResponseWriter, r *http.Request) {
+	http.Handle(fmt.Sprintf("%s/metrics", *routePrefix), promhttp.Handler())
+	http.HandleFunc(fmt.Sprintf("%s/probe", *routePrefix), func(w http.ResponseWriter, r *http.Request) {
 		sc.Lock()
 		conf := sc.C
 		sc.Unlock()
 		probeHandler(w, r, conf, logger, rh)
 	})
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	http.HandleFunc(fmt.Sprintf("%s/", *routePrefix), func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "text/html")
-		w.Write([]byte(`<html>
-    <head><title>Blackbox Exporter</title></head>
-    <body>
-    <h1>Blackbox Exporter</h1>
-    <p><a href="/probe?target=prometheus.io&module=http_2xx">Probe prometheus.io for http_2xx</a></p>
-    <p><a href="/probe?target=prometheus.io&module=http_2xx&debug=true">Debug probe prometheus.io for http_2xx</a></p>
-    <p><a href="/metrics">Metrics</a></p>
-    <p><a href="/config">Configuration</a></p>
-    <h2>Recent Probes</h2>
-    <table border='1'><tr><th>Module</th><th>Target</th><th>Result</th><th>Debug</th>`))
+		w.Write([]byte("<html>\n"))
+		w.Write([]byte("    <head><title>Blackbox Exporter</title></head>\n"))
+		w.Write([]byte("    <body>\n"))
+		w.Write([]byte("    <h1>Blackbox Exporter</h1>\n"))
+
+		w.Write([]byte(fmt.Sprintf("    <p><a href=\"%s/probe?target=prometheus.io&module=http_2xx\">Probe prometheus.io for http_2xx</a></p>\n", *routePrefix)))
```

Comment:
We could use relative paths here, the blackbox exporter doesn't do anything fancy.

Reply:
Great, since they are already relative urls I'll keep the code in this area the same then